setwd("/Users/student/Documents/GitHub/cubaMcavMsatSnp/shallowOnlyAnalysis/")
if (!require("pacman")) install.packages("pacman")
pacman::p_load("adegenet", "dendextend", "flextable", "ggdendro", "hierfstat", "Imap", "kableExtra", "paletteer", "patchwork", "officer", "poppr", "RColorBrewer", "reshape2", "StAMPP", "tidyverse", "vcfR", "vegan", "WGCNA")

cubaShallowVcf = read.vcfR("shallowOnly.vcf.gz") #This vcf file has all of your SNP genotype likelihoods and is generated by ANGSD
cubaShallowGenlight = vcfR2genlight(cubaShallowVcf, n.cores = 2) # Converts the vcf file into a file format that poppr uses the "genlight" format
locNames(cubaShallowGenlight) = paste(cubaShallowVcf@fix[,1],cubaShallowVcf@fix[,2],sep="_") 
westSites = c("Guanahacabibes", "IsladelaJuventud", "CayoJutias")
popDataShallow = read.table("shallowOnlyPopMap") # Reads in population data for each sample
popDataShallow$depth = factor("Shallow", levels = c("Shallow", "Mesophotic")) # Stratifies by depth zone
popDataShallow$location = factor(ifelse(popDataShallow$V2 %in% westSites, "Western", "Eastern"))


colnames(popDataShallow) = c("sample","population","depth", "location") 

#popDataShallow$depth[c(1:2)] = "Mesophotic"
#levels(popDataShallow$depth)

strata(cubaShallowGenlight) = data.frame(popDataShallow)
setPop(cubaShallowGenlight) = ~population

head(cubaShallowGenlight)

snpPca1Shallow = glPca(cubaShallowGenlight, nf = 75, n.cores = 1) 

snpPcaValuesShallow = as.data.frame(snpPca1Shallow$scores)
snpPcaShallow = data.frame(row.names = rownames(snpPcaValuesShallow), sample = rownames(snpPcaValuesShallow), 
                    site = popDataShallow$population, depth = popDataShallow$depth, PC1 = snpPcaValuesShallow[,1],
                    PC2 = snpPcaValuesShallow[,2])
head(snpPcaShallow)

# percentage of explained variance by axes
snpPcVarShallow = round(snpPca1Shallow$eig/sum(snpPca1Shallow$eig)*100, 1)

# merge centroid locations into ggplot dataframe
snpPcaShallow = merge(snpPcaShallow, aggregate(cbind(mean.x=PC1,mean.y=PC2)~site,snpPcaShallow, mean), by="site")

# SNP PCA biplot
display.brewer.pal(8, "Dark2")
brewer.pal(8, "Dark2")
plotPops = c("Cabo Lucrecia", "Cayo Anclitas", "Cayo Jut√≠as",
             "Cayo Sabinal", "Chivirico", "Guanahacabibes", "Isla de la Juventud")

cubasnpPcaShallowPlotA = ggplot(snpPcaShallow, aes(x = PC1, y = PC2, color = site, fill = site, shape = depth)) +
  geom_hline(yintercept = 0, color = "gray90", size = 0.5) +
  geom_vline(xintercept = 0, color = "gray90", size = 0.5) + 
  stat_ellipse(data = subset(snpPcaShallow, subset = !site %in% c("IsladelaJuventud")), type = "t", geom = "polygon", alpha = 0.1) + #ellipse
  geom_segment(data = subset(snpPcaShallow, subset = site %in% c("IsladelaJuventud")), aes(xend = mean.x, yend = mean.y, color = site)) + #spider
  geom_point(shape = 4, stroke=0.75, size = 3) + #individual's points indicated by circles
  geom_point(aes(x = mean.x, y = mean.y, shape = depth), size = 4, color = "black") + #population centroids indicated by triangles
  scale_shape_manual(values = c(21, 23), name = "Location") +
  scale_fill_manual(values=c("#D95F02", "#7570B3", "#E7298A", "#66A61E", "#E6AB02", "#A6761D", "#666666"), name = " Population", labels = plotPops) +
  scale_color_manual(values=c("#D95F02", "#7570B3", "#E7298A", "#66A61E", "#E6AB02", "#A6761D", "#666666"), name = " Population", labels = plotPops) +  
  xlab(paste ("PC 1 (", snpPcVarShallow[1],"%)", sep = "")) + #Prints percent variation explained by first axis
  ylab(paste ("PC 2 (", snpPcVarShallow[2],"%)", sep = "")) + #Prints percent variation explained by second axis
  guides(fill = guide_legend(override.aes = list(shape = 22, size = 4, color = NA), order = 1))+
  ggtitle("SNP") +
  theme_bw()

cubasnpPcaShallowPlot = cubasnpPcaShallowPlotA + 
  theme(axis.title.x = element_text(color = "black", size = 10),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.line.x = element_blank(),
        axis.title.y = element_text(color = "black", size = 10),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        axis.line.y = element_blank(),
        legend.position = "left",
        panel.border = element_rect(color = "black", size = 1.2),
        panel.background = element_rect(fill = "white"),
        plot.background = element_rect(fill = "white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())

cubasnpPcaShallowPlot

ggsave("cubasnpPcaShallowPlot.png", plot = cubasnpPcaShallowPlot, height = 20, width = 25, unit = "cm", dpi = 300)

## Calculating AMOVA from SNP data
cubaShallowVcf = read.vcfR("shallowOnly.vcf.gz") #This vcf file has all of your SNP genotype likelihoods and is generated by ANGSD
cubaShallowGenlight = vcfR2genlight(cubaShallowVcf, n.cores = 2) # Converts the vcf file into a file format that poppr uses the "genlight" format
#locNames(cubaShallowGenlight) = paste(cubaShallowVcf@fix[,1],cubaShallowVcf@fix[,2],sep="_") 
popDataShallow = read.table("shallowOnlyPopMap") # Reads in population data for each sample
popDataShallow$depth = factor("Shallow", levels = c("Shallow", "Mesophotic")) # Stratifies by depth zone
#popDataShallow$depth[c(1:2)] = "Mesophotic"
#levels(popDataShallow$depth)

colnames(popDataShallow) = c("sample","population","depth") 

strata(cubaShallowGenlight) = data.frame(popDataShallow)
setPop(cubaShallowGenlight) = ~population

head(cubaShallowGenlight)

#AMOVA
amovaShallow <- poppr.amova(cubaShallowGenlight, ~population) #Runs AMOVA
amovaShallow

set.seed(1999)
amovasignifShallow   <- randtest(amovaShallow, nrepet = 999) #Calculates significance levels of the AMOVA with 999 permutations
amovasignifShallow

#Pairwise Fst Map
set.seed(694)
cuba.fstShallow<-stamppFst(cubaShallowGenlight, nboots = 99, percent = 95, nclusters = 4) #99 permutations

cuba.fstShallow$Fsts
cuba.fstShallow$Pvalues

# reads in fst matrix
snpFstMaShallow = as.data.frame(cuba.fstShallow$Fsts)

pops = plotPops[c(6, 7, 2, 5, 1, 4, 3)]
names(snpFstMaShallow) = pops
row.names(snpFstMaShallow) = pops

snpFstMaShallow$Pop = factor(row.names(snpFstMaShallow), levels = unique(pops))
levels(snpFstMaShallow$Pop)

snpQMaShallow = data.frame(cuba.fstShallow$Pvalues)
names(snpQMaShallow)= pops
row.names(snpQMaShallow) = pops
snpQMaShallow$Pop = factor(row.names(snpQMaShallow), levels = unique(pops))

snpFstShallow = melt(snpFstMaShallow, id.vars = "Pop", value.name = "Fst", variable.name = "Pop2", na.rm = TRUE)
snpFstShallow = snpFstShallow[snpFstShallow$Pop != snpFstShallow$Pop2,]
snpFstShallow$Fst = round(snpFstShallow$Fst, 3)
snpFstShallow = snpFstShallow %>% mutate(Fst = replace(Fst, Fst < 0, 0))
head(snpFstShallow)

snpQShallow = melt(snpQMaShallow, id.vars = "Pop", value.name = "Pval", variable.name = "Pop2", na.rm = TRUE)
snpQShallow = snpQShallow[snpQShallow$Pop != snpQShallow$Pop2,]
snpQShallow$Qval = p.adjust(snpQShallow$Pval, method = "BH")
head(snpQShallow)


snpHeatmapAShallow = ggplot(data = snpFstShallow, aes(Pop, Pop2, fill = Fst))+ 
  geom_tile(color = "white")+ 
  scale_fill_gradient2(low = "white", high = "red", midpoint = 0, limit = c(0, 0.22), 
                       space = "Lab", name = expression(paste(italic("F")[ST])))+ 
  geom_text(data = snpFstShallow, aes(Pop, Pop2, label = Fst), color = ifelse(snpQShallow$Qval <= 0.05,"black", "darkgrey"), size = ifelse(snpQShallow$Qval < 0.05, 6, 5)) +
  guides(fill=guide_colorbar(barwidth = 1, barheight = 12, title.position = "top", title.hjust = 0.5))+ 					
  scale_y_discrete(position = "right")+
  scale_x_discrete(labels = str_wrap(snpFstShallow$Pop, width = 6)) +
  ggtitle("   SNP") +
  theme_minimal()

snpHeatmapShallow = snpHeatmapAShallow + theme(
  axis.text.x = element_text(vjust = 1, size = 16, hjust = 0.5, color = "black"), 
  axis.text.y = element_text(size = 16, color = "black"),
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  legend.justification = c(1, 0),
  legend.position = "right",
  legend.direction = "vertical",
  legend.title = element_text(size = 16),
  legend.text = element_text(size = 14),
  plot.title = element_text(size = 16)
)

snpHeatmapShallow

ggsave("snpHeatmapShallow.png", plot = snpHeatmapShallow, height = 10, width = 25, unit = "cm", dpi = 300)

